content/unpack.md

```diff
@@ -0,0 +1,1084 @@
+---
+layout: sip
+permalink: /sips/:title.html
+stage: implementation
+status: under-review
+title: SIP-XX - Unpack Case Classes into Parameter Lists and Argument Lists
+---
+
+**By: Li Haoyi**
+
+## History
+
+| Date        | Version            |
+|-------------|--------------------|
+| 23 Aug 2024 | Initial Draft      |
+
+
+## Summary
+
+This proposal provides a syntax to "unpack" a `case class` _type_ into a definition-site 
+parameter list via the `unpack` keyword, and to "unpack" `case class` _value_ into a
+definition-site argument list via `*`: 
+
+```scala
+case class RequestConfig(url: String, 
+                         connectTimeout: Int,
+                         readTimeout: Int)
+
+def downloadSimple(unpack config: RequestConfig) = doSomethingWith(config)
+def downloadAsync(unpack config: RequestConfig, ec: ExecutionContext) = doSomethingWith(config)
+def downloadStream(unpack config: RequestConfig) = doSomethingWith(config)
```

Comment:
Personal opinion: the `doSomethingWith()`s here and elsewhere are introducing more confusion than benefit.  While I had a suspicion where you were going in this proposal, this example just leaves me going, "So what?  That's exactly the same as today."

(Continued below...)